sites/funcampus/src/backend/base/cache.py

```diff
@@ -2,81 +2,42 @@
-        return response
+    max_age = get_max_age(response)
+    if max_age is not None and max_age > max_ttl:
+        if response.has_header("Expires"):
+            # Remove the Expires response Header because patch_response_headers()
+            # adds it only if it isn't already setI
+            del response["Expires"]
+        patch_response_headers(response, cache_timeout=max_ttl)
 
-    # This *must* be TZ-aware
-    timestamp = now()
 
-    placeholders = toolbar.content_renderer.get_rendered_placeholders()
-    # Checks if there's a plugin using the legacy "cache = False"
-    placeholder_ttl_list = []
-    vary_cache_on_set = set()
-    for ph in placeholders:
-        # get_cache_expiration() always returns:
-        #     EXPIRE_NOW <= int <= MAX_EXPIRATION_IN_SECONDS
-        ttl = ph.get_cache_expiration(request, timestamp)
-        vary_cache_on = ph.get_vary_cache_on(request)
-
-        placeholder_ttl_list.append(ttl)
-        if ttl and vary_cache_on:
-            # We're only interested in vary headers if they come from
-            # a cache-able placeholder.
-            vary_cache_on_set |= set(vary_cache_on)
-
-    if EXPIRE_NOW not in placeholder_ttl_list:
-        if placeholder_ttl_list:
-            min_placeholder_ttl = min(x for x in placeholder_ttl_list)
-        else:
-            # Should only happen when there are no placeholders at all
-            min_placeholder_ttl = MAX_EXPIRATION_TTL
-        ttl = min(get_cms_setting("CACHE_DURATIONS")["content"], min_placeholder_ttl)
-
-        if ttl > 0:
-            # Adds expiration, etc. to headers
-            patch_response_headers(response, cache_timeout=ttl)
-            patch_vary_headers(response, sorted(vary_cache_on_set))
+def details(request, *args, **kwargs):
+    """
+    This method redefines cms.views.details view.
+    It allows to set a maximum TTL for the browser cache response header.
+    See the setting CMS_MAX_BROWSER_CACHE_TTL for more information.
+    """
 
-            version = _get_cache_version()
-            # We also store the absolute expiration timestamp to avoid
-            # recomputing it on cache-reads.
-            expires_datetime = timestamp + timedelta(seconds=ttl)
-            cache.set(
-                _page_cache_key(request),
-                (response.content, response._headers, expires_datetime,),
-                ttl,
-                version=version,
-            )
-            # See note in invalidate_cms_page_cache()
-            _set_cache_version(version)
+    response = django_cms_details(request, *args, **kwargs)
+    # The response return by cms.views.detail is either a TemplateResponse or a BaseHttpResponse
+    # In the case of a TemplateResponse, the rendering can be delayed and the cache headers not
+    # set yet. That's why we have to add a post rendering-callback to patch the response headers.
+    if hasattr(response, "is_rendered") and not response.is_rendered:
```

Comment:
I'm wondering, since what you need to do is to modify a header on the response, why not do it in a small middleware that we add at the last position of our middlewares!?

Reply:
I considered this middleware solution, but what bothered me is that a middleware acts on all responses.
How to target only the one returned by Django CMS ?

I didn't want to introduce a middleware that could disturb other django applications.

Reply:
I would say that the cache approach in Django is anyway across the site so it makes sense to centralize all modifications to a middleware? In our case this site will only be DjangoCMS so if we apply this rule in this repository it seems ok?
I think it's worth avoiding the monkey patch if we can :sweat_smile:

Reply:
I understand your point of view. I'm just not sure about the durability of this statement «this site will only be DjangoCMS» :smile: 

I'll update my PR to do this with a middleware.